pkg/client/cache/delta_fifo.go

```diff
@@ -522,6 +522,23 @@ func (f *DeltaFIFO) syncKey(key string) error {
 		return nil
 	}
 
+	// If we are doing Resync() and there is already an event queued for that object,
```

Comment:
I'm worried about all the places that will now get a new event type that didn't before.